src/acl/ConnMark.cc

```diff
@@ -19,99 +19,87 @@
 #include "HttpRequest.h"
 #include "parser/Tokenizer.h"
 #include "Parsing.h"
+#include "sbuf/Stream.h"
 
 #include <string>
 
-ACLConnMark::ACLConnMark() {}
-
-ACL *
-ACLConnMark::clone() const
-{
-    return new ACLConnMark(*this);
-}
-
 bool
-ACLConnMark::empty () const
+Acl::ConnMark::empty () const
 {
     return false;
 }
 
 void
-ACLConnMark::parse()
+Acl::ConnMark::parse()
 {
     while (const char *t = ConfigParser::strtokFile()) {
         SBuf token(t);
-        if (token.isEmpty()) {
-            continue;
-        }
-
-        nfmark_t _mask = 0xffffffff;
-        nfmark_t _mark = 0;
+        nfmark_t mask = 0xffffffff;
+        nfmark_t mark = 0;
 
         Parser::Tokenizer tokenizer(token);
-        int64_t tmp_result;
+        int64_t number;
         // get mark
-        if (tokenizer.int64(tmp_result, 0, false))
-            _mark = static_cast<nfmark_t>(tmp_result);
+        if (tokenizer.int64(number, 0, false))
+            mark = static_cast<nfmark_t>(number);
         else {
-            debugs(28, DBG_CRITICAL, MYNAME << ": bad mark '" << token << "'");
-            self_destruct();
+            throw TexcHere(ToSBuf("acl ", typeString(), ": bad mark value ", token));
         }
 
-        // mark with mask ?
-        SBuf::size_type pos = token.find('/');
-        if (pos != SBuf::npos) {
-            token = SBuf(t + pos + 1);
-            tokenizer = Parser::Tokenizer(token);
-            if (tokenizer.int64(tmp_result, 0, false))
-                _mask = static_cast<nfmark_t>(tmp_result);
-            else {
-                debugs(28, DBG_CRITICAL, MYNAME << ": bad mask '" << token << "'");
-                self_destruct();
+        if (!tokenizer.atEnd()) {
+            if (tokenizer.skip('/')) {
+                if (tokenizer.int64(number, 0, false))
+                    mask = static_cast<nfmark_t>(number);
```

Comment:
AFAICT, this code silently ignores user input due to a 64- to 32-bit conversion. Before you cast, add an overflow check. For example:

```
if (number > std::numeric_limits<nfmark_t>::max())
    throw ... too big
```

BTW, you may want to encapsulate all those repeated number parsing and checks into a helper function, but I do not insist on that.